pkg/receive/handler.go

```diff
@@ -324,29 +326,39 @@ func (h *Handler) forward(ctx context.Context, tenant string, r replica, wreq *p
 	}
 	h.mtx.RUnlock()
 
-	return h.parallelizeRequests(ctx, tenant, replicas, wreqs)
+	n, ec := h.parallelizeRequests(ctx, tenant, replicas, wreqs)
```

Comment:
why not just `for err := range <-ec`? without n?

Reply:
we need to know the potential number of results so we know when we have reached quorum, it could also return the quorum amount instead, but that would really result in the same code